tests/e2e/gateway/03-assert.yaml

```diff
@@ -0,0 +1,106 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: query-frontend
+    app.kubernetes.io/instance: foo
+    app.kubernetes.io/managed-by: tempo-operator
+    app.kubernetes.io/name: tempo
+  name: tempo-foo-query-frontend
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: query-frontend
+      app.kubernetes.io/instance: foo
+      app.kubernetes.io/managed-by: tempo-operator
+      app.kubernetes.io/name: tempo
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: query-frontend
+        app.kubernetes.io/instance: foo
+        app.kubernetes.io/managed-by: tempo-operator
+        app.kubernetes.io/name: tempo
+        tempo-gossip-member: "true"
+    spec:
+      containers:
+      - args:
+        - -target=query-frontend
+        - -config.file=/conf/tempo-query-frontend.yaml
+        - -mem-ballast-size-mbs=1024
+        - -log.level=info
+        - --storage.trace.s3.secret_key=$(S3_SECRET_KEY)
+        - --storage.trace.s3.access_key=$(S3_ACCESS_KEY)
+        env:
+        - name: S3_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              key: access_key_secret
+              name: minio-test
+        - name: S3_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              key: access_key_id
+              name: minio-test
+        image: docker.io/grafana/tempo:2.3.0
```

Comment:
can we make this version independent? I would also remove what is not intended to be tested.